robustifier/src/main/kotlin/cmu/isr/tolerance/Utils.kt

```diff
@@ -0,0 +1,139 @@
+import cmu.isr.ts.LTS
+import cmu.isr.ts.MutableDetLTS
+import cmu.isr.ts.MutableLTS
+import cmu.isr.ts.alphabet
+import cmu.isr.ts.lts.CompactDetLTS
+import cmu.isr.ts.lts.asLTS
+import cmu.isr.ts.lts.CompactLTS
+import cmu.isr.ts.lts.checkSafety
+import cmu.isr.ts.lts.makeErrorState
+import cmu.isr.ts.nfa.determinise
+import net.automatalib.automata.fsa.impl.compact.CompactDFA
+import net.automatalib.util.automata.builders.AutomatonBuilders
+import net.automatalib.words.Alphabet
+import java.util.*
+
+/**
+ * Finds all transitions in T
+ */
+fun ltsTransitions(T : LTS<Int, String>) : Set<Triple<Int,String,Int>> {
+    val init = T.initialStates
+    if (init.isEmpty()) {
+        return emptySet()
+    }
+
+    // BFS for all transitions
+    val transitions = mutableSetOf<Triple<Int,String,Int>>()
+    val visited : MutableSet<Int> = mutableSetOf()
+    val q : Queue<Int> = LinkedList()
+    q.addAll(init)
+    while (q.isNotEmpty()) {
+        val state = q.remove()
+        if (visited.contains(state)) {
+            continue
+        }
+        visited.add(state)
+        for (a in T.alphabet()) {
+            val adjacent = T.getTransitions(state, a)
+            for (dst in adjacent) {
+                transitions.add(Triple(state, a, dst))
+                q.add(dst)
+            }
+        }
+    }
+
+    return transitions
+}
+
+/**
+ * It's ridiculous how tough they make it to copy.
+ * The algorithm makes a key assumption: that initial states are the lowest state ID's, and will be added to newLTS
+ * with the same lowest state IDs.
+ */
+fun copyLTS(T : CompactDetLTS<String>) : CompactDetLTS<String> {
+    val newLTS = AutomatonBuilders.newDFA(T.inputAlphabet)
+        .withInitial(T.initialStates)
```

Comment:
If T is a DetLTS (i.e., a DFA), then there should be only one initial state. So you can use T.initialState